src/components/App/index.jsx

```diff
@@ -1,8 +1,173 @@
+    });
+
+    this.getWinner(boardCopy, x, y);
+  };
+
+  verticalCheck(board, x, y) {
+    const { currentTile } = this.state;
+
+    // down
+    for (let j = y - 3; j <= y; j++) {
+      if (board[x][j] !== currentTile) {
+        return;
+      }
+    }
+
+    alert("we have a winner!");
+    this.setState({ currentTile: "red", board: [[], [], [], [], [], [], []] });
+  }
+
+  horizontalCheck(board, x, y) {
+    const { currentTile } = this.state;
+    let firstIndex = x;
+    let lastIndex = x;
+
+    // left
+    for (let i = x - 1; i >= 0; i--) {
+      if (!board[i][y] || board[i][y] !== currentTile) {
+        break;
+      } else {
+        firstIndex = i;
+      }
+    }
+
+    // right
+    for (let j = x + 1; j < board.length; j++) {
+      if (!board[j][y] || board[j][y] !== currentTile) {
+        break;
+      } else {
+        lastIndex = j;
+      }
+    }
+
+    if (lastIndex - firstIndex === 3) {
+      alert("we have a winner!");
+      this.setState({
+        currentTile: "red",
+        board: [[], [], [], [], [], [], []],
+      });
+    }
+  }
+
+  diagonalCheck1(board, x, y) {
+    const { currentTile } = this.state;
+    const colLength = 5;
+
+    let firstIndex = x;
+    let lastIndex = x;
+
+    for (
+      let i = x + 1, j = y + 1;
+      i < board.length && j <= colLength;
+      i++, j++
+    ) {
+      if (!board[i][j] || board[i][j] !== currentTile) {
+        break;
+      } else {
+        lastIndex = i;
+      }
+    }
+
+    for (let i = x - 1, j = y - 1; i >= 0 && j >= 0; i--, j--) {
+      if (!board[i][j] || board[i][j] !== currentTile) {
+        break;
+      } else {
+        firstIndex = i;
+      }
+    }
+
+    if (Math.abs(lastIndex - firstIndex) === 3) {
+      alert("we have a winner!");
+      this.setState({
+        currentTile: "red",
+        board: [[], [], [], [], [], [], []],
+      });
+    }
+  }
+
+  diagonalCheck2(board, x, y) {
+    const { currentTile } = this.state;
+    const colLength = 5;
+
+    let firstIndex = x;
+    let lastIndex = x;
+
+    for (let i = x - 1, j = y + 1; i >= 0 && j <= colLength; i--, j++) {
+      if (!board[i][j] || board[i][j] !== currentTile) {
+        break;
+      } else {
+        firstIndex = i;
+      }
+    }
+
+    for (let i = x + 1, j = y - 1; i < board.length && j >= 0; i++, j--) {
+      if (!board[i][j] || board[i][j] !== currentTile) {
+        break;
+      } else {
+        lastIndex = i;
+      }
+    }
+
+    if (Math.abs(lastIndex - firstIndex) === 3) {
+      alert("we have a winner!");
+      this.setState({
+        currentTile: "red",
+        board: [[], [], [], [], [], [], []],
+      });
+    }
+  }
+
+  getWinner(board, x, y) {
+    this.verticalCheck(board, x, y);
+    this.horizontalCheck(board, x, y);
+    this.diagonalCheck1(board, x, y);
+    this.diagonalCheck2(board, x, y);
+  }
+
   render() {
-    return <div>App</div>;
+    const { board } = this.state;
+    const cells = [];
```

Comment:
you probably want to call this boardElem or something like that. Cells sounds smaller than rows because a row contains cells. In your case, your cells are going to contain rows. The naming is a little confusing right now.